src/main/java/Duke.java

```diff
@@ -1,10 +1,232 @@
+                throw new DukeException("Hold up! Description cannot be empty!");
+            }
+            if (date.length < 2 || date[1].trim().length() == 0) {
+                throw new DukeException("Wait! When is this event??");
+            }
+            Event newEvent = new Event(description.substring(0, description.indexOf("/at") - 1),
+                    description.substring(description.indexOf("/") + 4));
+            tasks.add(numOfTasks, newEvent);
+            numOfTasks++;
+        } else {
+            echo(input);
+        }
+        saveToHardDisk();
+        System.out.println("Got it. I've added this task:\n" + "  " + tasks.get(numOfTasks - 1).toString());
+        System.out.println("Now you have " + numOfTasks + taskTense() + " in the list.");
+    }
+
+    private void deleteTask(int index) throws DukeException {
+        try {
+            Task removedTask = tasks.remove(index);
+            System.out.println("Done! " + removedTask.toString() + " has been deleted :(");
+            numOfTasks--;
+            saveToHardDisk();
+        } catch (Exception e) {
+            throw new DukeException("I can't find such a task to delete!");
+        }
+    }
+
+    private void saveToHardDisk() throws DukeException {
+        byte[] data = convertTasksToListString().getBytes();
+        Path p = Paths.get("./data/duke.txt");
+        Path dataPath = Paths.get("./data");
+
+        boolean isDataFileExisting = Files.exists(dataPath);
+
+        if (!isDataFileExisting) {
+            File f1 = new File(String.valueOf(dataPath));
+            isDataFileExisting = f1.mkdir();
+        }
+        if (isDataFileExisting) {
+            try (OutputStream out = new BufferedOutputStream(
+                    Files.newOutputStream(p))) {
+                out.write(data, 0, data.length);
+            } catch (IOException x) {
+                throw new DukeException("There was an error saving your changes!");
+            }
+        }
+    }
+
+    private String convertTasksToListString() {
+        StringBuilder list = new StringBuilder();
+        for (int i = 0; i < numOfTasks; i++) {
+            String newLine = (i + 1) + ". " + tasks.get(i).toString() + "\n";
+            list.append(newLine);
+        }
+        return list.toString();
+    }
+
+    private ArrayList<Task> readTasksFromHardDrive() {
+        Path p = Paths.get("./data/duke.txt");
+        boolean isFileExisting = Files.exists(p);
+        ArrayList<Task> tasks = new ArrayList<>();
+
+        if (!isFileExisting) {
+            return tasks;
+        }
+        try (InputStream in = Files.newInputStream(p);
+             BufferedReader reader =
+                     new BufferedReader(new InputStreamReader(in))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                Task newTask;
+                // process data
+                switch (line.charAt(4)) {
```

Comment:
Could the `4` be a static variable? This could reduce the reliance on 'magic numbers'